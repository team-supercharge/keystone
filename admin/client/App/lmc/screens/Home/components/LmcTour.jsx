import React, { Component } from 'react';
import introJs from 'intro.js';
import {
    ResponsiveText,
    GlyphButton,
} from '../../../../elemental';

class LmcTour extends Component {

    startTour() {
        this.props.onStart();
        introJs()
            .setOption('exitOnOverlayClick', true)
            .setOption('scrollToElement', true)
            .setOption('hideNext', true)
            .setOption('disableInteraction', true)
            .onskip(function () {
                console.log('onbeforeexit')
            })
            .addStep({
                element: document.querySelectorAll('#intro-js-step-dashboard')[0],
                intro: TOUR_HELLO,
                position: 'top',
            })
            .addStep({
                element: document.querySelectorAll('#intro-js-step-residents')[0],
                intro: TOUR_RESIDENTS,
                position: 'right',
            })
            .addStep({
                element: document.querySelectorAll('#intro-js-step-add-residents')[0],
                intro: TOUR_ADD_RESIDENTS,
                position: 'right',
            })
            .addStep({
                element: document.querySelectorAll('#intro-js-step-carers')[0],
                intro: TOUR_CARERS,
                position: 'top',
            })
            .addStep({
                element: document.querySelectorAll('#intro-js-step-add-carers')[0],
                intro: TOUR_CARERS_INVITE,
                position: 'top',
            })
            .addStep({
                element: document.querySelectorAll('#intro-js-step-incidents')[0],
                intro: TOUR_INCIDENTS,
                position: 'left',
            })
            .addStep({
                element: document.querySelectorAll('#intro-js-step-tasks')[0],
                intro: TOUR_TASKS,
                position: 'left',
            })
            .addStep({
                element: document.querySelectorAll('#intro-js-step-tasks-info')[0],
                intro: TOUR_TASKS_INFO,
                position: 'top',
            })
            .addStep({
                element: document.querySelectorAll('#intro-js-step-advert')[0],
                intro: TOUR_ADVERT,
                position: 'left',
            })
            .addStep({
                element: document.querySelectorAll('#intro-js-step-navbar')[0],
                intro: TOUR_NAVBAR,
                position: 'bottom',
            })
            .addStep({
                element: document.querySelectorAll('#intro-js-done')[0],
                intro: TOUR_DONE,
                hideNext: true,
                position: 'top',
            })
            .start();
    }

    componentWillUnmount() {
        introJs().exit();
    }

    render() {
        return (
            <div style={{ paddingTop: 20, width: 200 }}>
                <GlyphButton
                    block
                    color="success"
                    glyph="play"
                    onClick={this.startTour.bind(this)}
                    position="left"
                    title={`Start Tour`}
                >
                    <ResponsiveText
                        visibleSM="Start Tour"
                        visibleMD="Start Tour"
                        visibleLG={`Start Tour`}
                    />
                </GlyphButton>
            </div>
        );
    }
}

const TOUR_HELLO = 'On the home page you\'ll see a snapshot of what has been happening over the last day and can perform some quick actions.';
const TOUR_RESIDENTS = 'Up here you\'ll see your active residents, these are those that are currently in your home';
const TOUR_ADD_RESIDENTS = 'The first step is to add your residents and assign them to your care team. This information then gets sent to the Carer App and allows carers to record care on the go.'
const TOUR_CARERS = 'Down here you can see who from your care team has been online and logged care today.';
const TOUR_CARERS_INVITE = 'You can also invite other members of your team to join you in Log my Care.';
const TOUR_INCIDENTS = 'Next is your incidents list. This gets populated when your carers record an incident log on the Carer App and lets you know what you need to deal with.';
const TOUR_TASKS = 'The last section shows you the status of today’s To-Do\'s and Logs. This is where your scheduled care tasks and ad-hoc quick-logs recorded by your care team show up. ';
const TOUR_TASKS_INFO = 'You can instantly see how many To-Do\'s are overdue, pending and have been completed today.';
const TOUR_ADVERT = 'On the right you can see our tips and tricks videos to help you make the most of Log my Care along with our latest offers.';
const TOUR_NAVBAR = 'Along the top you\'ll see menu options for all features, including how to access our help centre.';
const TOUR_DONE = 'That\'s it - you\'ve finished the walkthrough of the Care Office!';


export default LmcTour;
