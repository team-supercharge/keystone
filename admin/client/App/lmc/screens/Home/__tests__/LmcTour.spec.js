import React from 'react'
import { shallow } from 'enzyme'
import LmcTour from '../components/LmcTour'

describe('LmcTour', () => {
    let wrapper

    beforeEach(() => {
        wrapper = shallow(<LmcTour/>)
    })

    test('should render correct snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })
})