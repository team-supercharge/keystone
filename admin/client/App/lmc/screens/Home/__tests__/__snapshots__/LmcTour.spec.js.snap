// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LmcTour should render correct snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    style={
      Object {
        "paddingTop": 20,
        "width": 200,
      }
    }
  >
    <GlyphButton
      block={true}
      color="success"
      glyph="play"
      glyphStyle={Object {}}
      onClick={[Function]}
      position="left"
      title="Start Tour"
    >
      <ResponsiveText
        component="span"
        visibleLG="Start Tour"
        visibleMD="Start Tour"
        visibleSM="Start Tour"
      />
    </GlyphButton>
  </div>,
  "nodes": Array [
    <div
      style={
        Object {
          "paddingTop": 20,
          "width": 200,
        }
      }
    >
      <GlyphButton
        block={true}
        color="success"
        glyph="play"
        glyphStyle={Object {}}
        onClick={[Function]}
        position="left"
        title="Start Tour"
      >
        <ResponsiveText
          component="span"
          visibleLG="Start Tour"
          visibleMD="Start Tour"
          visibleSM="Start Tour"
        />
      </GlyphButton>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <LmcTour />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": LmcTour {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "startTour": [Function],
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          style={
            Object {
              "paddingTop": 20,
              "width": 200,
            }
          }
        >
          <GlyphButton
            block={true}
            color="success"
            glyph="play"
            glyphStyle={Object {}}
            onClick={[Function]}
            position="left"
            title="Start Tour"
          >
            <ResponsiveText
              component="span"
              visibleLG="Start Tour"
              visibleMD="Start Tour"
              visibleSM="Start Tour"
            />
          </GlyphButton>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div
          style={
            Object {
              "paddingTop": 20,
              "width": 200,
            }
          }
        >
          <GlyphButton
            block={true}
            color="success"
            glyph="play"
            glyphStyle={Object {}}
            onClick={[Function]}
            position="left"
            title="Start Tour"
          >
            <ResponsiveText
              component="span"
              visibleLG="Start Tour"
              visibleMD="Start Tour"
              visibleSM="Start Tour"
            />
          </GlyphButton>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <LmcTour />,
}
`;
